Dockerfile

```diff
@@ -0,0 +1,14 @@
+# This Dockerfile is used by CI to publish openshift/odo:binary-artifacts
+# It builds an image containing the Mac, Win and Linux version of odo binary on the
+# OpenShift golang image.
+
+FROM registry.svc.ci.openshift.org/openshift/release:golang-1.11 AS builder
+COPY . /go/src/github.com/openshift/odo
+WORKDIR /go/src/github.com/openshift/odo
+RUN go get github.com/mitchellh/gox &&\
+    make cross
+
+FROM registry.svc.ci.openshift.org/ocp/4.2:cli
```

Comment:
@mohammedzee1000 I have updated the file as per your comment, but getting unauthorized issue on pulling this image. So can you try the same docker build form your end once.
```
Step 6/9 : FROM registry.svc.ci.openshift.org/ocp/4.2:cli
unauthorized: authentication required
```

Reply:
rename this to Dockerfile.rhel
Make another copy where you use a different image, like maybe uni

Reply:
Then test with that image. Dockerfile.rhel should be built internally

Reply:
$ vi ~/.docker/config.json
```
{"auths":{},"credsStore":"desktop","stackOrchestrator":"swarm"}
```
@mohammedzee1000 Probably blank auth causing the issue ?

Reply:
$ docker pull registry.svc.ci.openshift.org/ocp/4.2:cli
Error response from daemon: unauthorized: authentication required

Reply:
> rename this to Dockerfile.rhel
Make another copy where you use a different image, like maybe ubi
registry.access.redhat.com/ubi7/ubi

Do we really need this ?

Reply:
We need a way to check if the container works correctly, so do it for now. We can figure out the rest once we have something that works

Reply:
I believe that image is hidden behind a firewall. Let people @jwforres get online, then we will know what more to change

Reply:
@mohammedzee1000 It works with the registry image ```registry.access.redhat.com/ubi7/ubi```

```
$ docker build .
Sending build context to Docker daemon  514.7MB
Step 1/9 : FROM registry.svc.ci.openshift.org/openshift/release:golang-1.11 AS builder
 ---> c24f8126dfcb
Step 2/9 : COPY . /go/src/github.com/openshift/odo
 ---> d06eed57c59c
Step 3/9 : WORKDIR /go/src/github.com/openshift/odo
 ---> Running in 03239e063fec
Removing intermediate container 03239e063fec
 ---> 72d8c4bd551c
Step 4/9 : RUN go get github.com/mitchellh/gox &&    make cross
 ---> Running in 9d66147b9336
gox -osarch="darwin/amd64 linux/amd64 windows/amd64" -output="dist/bin/{{.OS}}-{{.Arch}}/odo" -ldflags="-w -X github.com/openshift/odo/pkg/odo/cli/version.GITCOMMIT=c3df454" ./cmd/odo/
Number of parallel builds: 3

-->    darwin/amd64: github.com/openshift/odo/cmd/odo
-->     linux/amd64: github.com/openshift/odo/cmd/odo
-->   windows/amd64: github.com/openshift/odo/cmd/odo
Removing intermediate container 9d66147b9336
 ---> e2b1cd690ce5
Step 5/9 : FROM registry.access.redhat.com/ubi7/ubi
latest: Pulling from ubi7/ubi
2b10c4b8508a: Already exists 
1fe78366afe1: Already exists 
Digest: sha256:adf499ed354847d81fbd993546c8bd1ba414e3a45fa10b5c68f532fa09d2f172
Status: Downloaded newer image for registry.access.redhat.com/ubi7/ubi:latest
 ---> 247ee58855fd
Step 6/9 : LABEL Description="This image is used to keep odo(Developer-focused CLI for OpenShift) executables for Windows, macOS and linux" Version="1.0"
 ---> Running in 651341a54bc6
Removing intermediate container 651341a54bc6
 ---> e31bf51309dd
Step 7/9 : COPY --from=builder /go/src/github.com/openshift/odo/dist/bin/darwin-amd64/odo /usr/share/openshift/odo/mac/odo
 ---> ed1d4f7f9eac
Step 8/9 : COPY --from=builder /go/src/github.com/openshift/odo/dist/bin/windows-amd64/odo.exe /usr/share/openshift/odo/windows/odo.exe
 ---> 6740927c2021
Step 9/9 : COPY --from=builder /go/src/github.com/openshift/odo/dist/bin/linux-amd64/odo /usr/share/openshift/odo/linux/odo
 ---> 1066155e15be
Successfully built 1066155e15be
```